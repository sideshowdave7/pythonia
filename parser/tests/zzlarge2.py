[1, 2, 3, [4, 5]][2][1]
def a():
 pass
def b(func):
 j = 3
 k = []
 for i in range(10):
  k.append(func(i, j))
b(lambda x, y: x*y)
y = lambda c: 1, 2, c, 3
z = lambda d: lambda x, y: x*y*d
b(z(3))
omega = {"a", "b",}
decy = {"a":"b", "c":"d", "e":"f", 1:3,}
y = [1,2,3,4,5,]
labber = [1,2,3,[4,5,6]][3][2]
def c(nammer): 3 if nammer == 3 else 2
c(z(3)(1,2,))
omega2 = {}
omega3 = {1, 2, 3, 4, 5, 6, [1, 2, 3,]}
global a
global a, b, c
global a, b, c, d
nonlocal q
nonlocal r, v
nonlocal w, x, y, z
def main():
 while True:
  pass
 else:
  y = 54*2**2/7
 while False:
  z = 4+3
 while 4 == x:
  x -= 2/8
 for i in range(10):
  pass
 def moar(b,):
  for i in [1, 2, 3, 4]:
   print(i)
if __name__ == "__main__":
 main()

