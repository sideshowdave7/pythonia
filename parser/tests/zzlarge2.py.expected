(program
 (expr (indexed (list 1 2 3 (list 4 5)) (subscript 2) (subscript 1)))
 (def (a) (suite (pass)))
 (def
  (b func)
  (suite
   (= (j) 3)
   (= (k) (list))
   (for
    i
    (indexed range (called 10))
    (suite
     (expr (indexed k (dot append) (called (indexed func (called i j)))))))))
 (expr (indexed b (called (lambda (x y) (term x ("*" y))))))
 (= (y) (tuple (lambda (c) 1) 2 c 3))
 (= (z) (lambda (d) (lambda (x y) (term x ("*" y) ("*" d)))))
 (expr (indexed b (called (indexed z (called 3)))))
 (= (omega) (set "a" "b"))
 (= (decy) (dict ("a" "b") ("c" "d") ("e" "f") (1 3)))
 (= (y) (list 1 2 3 4 5))
 (= (labber) (indexed (list 1 2 3 (list 4 5 6)) (subscript 3) (subscript 2)))
 (def (c nammer) (expr (if (comparison nammer ("==" 3)) 3 2)))
 (expr (indexed c (called (indexed z (called 3) (called 1 2)))))
 (= (omega2) (dict))
 (= (omega3) (set 1 2 3 4 5 6 (list 1 2 3)))
 (global a)
 (global a b c)
 (global a b c d)
 (nonlocal q)
 (nonlocal r v)
 (nonlocal w x y z)
 (def
  (main)
  (suite
   (while
    True
    (suite (pass))
    (suite (= (y) (term 54 ("*" (power 2 2)) ("/" 7)))))
   (while False (suite (= (z) (arith 4 ("+" 3)))))
   (while (comparison 4 ("==" x)) (suite ("-=" (x) (term 2 ("/" 8)))))
   (for i (indexed range (called 10)) (suite (pass)))
   (def
    (moar b)
    (suite (for i (list 1 2 3 4) (suite (expr (indexed print (called i)))))))))
 (cond
  ((comparison __name__ ("==" "__main__"))
   (suite (expr (indexed main (called)))))))

